rewrite-core/src/main/java/org/openrewrite/Parser.java

```diff
@@ -97,6 +97,22 @@ default Parser<S> reset() {
         return this;
     }
 
+    /**
+     * Returns {@link java.nio.charset.StandardCharsets#UTF_8} unless the ctx is an instance of
+     * {@link ParsingExecutionContextView} in which case it will return the charset from
+     * the {@link ParsingExecutionContextView#getCharset()}
+     */
+    default Charset getCharset(ExecutionContext ctx) {
+        if (ctx instanceof ParsingExecutionContextView) {
+            ParsingExecutionContextView parsingExecutionContextView = (ParsingExecutionContextView) ctx;
+            if (parsingExecutionContextView.getCharset() != null) {
+                return parsingExecutionContextView.getCharset();
+            }
+        }
+        return StandardCharsets.UTF_8;
```

Comment:
This isn't quite right. These "views" of execution context are wrappers/decorators of the underlying execution context that provide type-safe access to particular key-value pairs. An execution context view has no state of its own.

The only reason to `instanceof` check an execution context is to avoid needlessly re-wrapping

Reply:
thanks pushed the fix that does make sense since the delegating execution context will get the `org.openrewrite.parser.charset` property value.